### PR TITLE
[Feat] Add list toggle to filters

### PR DESCRIPTION
**Pull request by igorDykhta**
- create a AddByDatasetButton
- Display flter by list or dataset
- Add filter button shows typeahead and dataset selection
- Move add buttons to right of title to filter / base map panel

Signed-off-by: Ihor Dykhta <dikhta.igor@gmail.com>

### CODE REVIEW

File: src/components/src/side-panel/side-bar.tsx [hunk 1]
```diff
@@ -24,14 +24,14 @@ import {ArrowRight} from '../common/icons';
 
 type CollapseButtonProps = {
   isOpen: boolean;
-  onClick: () => void;
+  onClick: (e: React.MouseEvent<HTMLDivElement>) => void;
 };
 
 type SideBarProps = {
```
Comment (heshan0131) [inline src/components/src/side-panel/side-bar.tsx:30]: this type should be exported from `components/index.ts`

File: src/components/src/side-panel/panel-tab.tsx [hunk 2]
```diff
@@ -37,7 +37,7 @@ export type PanelItem = {
 type PanelTabProps = {
```
Comment (heshan0131) [inline src/components/src/side-panel/panel-tab.tsx:37]: this type should be exported from components/index.ts

File: src/components/src/index.ts [hunk 3]
```diff
@@ -100,6 +100,7 @@ export {default as DatasetTagFactory} from './side-panel/common/dataset-tag';
 
 export {default as FilterManagerFactory} from './side-panel/filter-manager';
 export {default as FilterPanelFactory} from './side-panel/filter-panel/filter-panel';
+export {default as AddFilterButtonFactory} from './side-panel/filter-panel/add-filter-button';
```
Comment (heshan0131) [inline src/components/src/index.ts:103]: missing some type export here:

```
MapStyleSelectorProps,
LayerGroupSelectorProps,
MapManagerProps,
SourceDataCatalogProps,
SourceDataSelectorProps,
DatasetTitleProps,
DatasetInfoProps,
DatasetTagProps,
CustomPanelsProps
```

File: src/components/src/side-panel/common/types.ts [hunk 4]
```diff
@@ -1,8 +1,21 @@
...
+};
+
+export type PanelMeta = {
+  id: string;
+  label: string;
+  iconComponent: React.ElementType;
+  onClick: null;
+};
+
 export type DatasetInfoProps = {
```
Comment (heshan0131) [inline src/components/src/side-panel/common/types.ts:19]: this should be 
`dataset: MiniDataset`?

**Review by heshan0131 — APPROVED** [review]

File: src/components/src/side-panel/common/types.ts (hunk 4 above)
Comment (igorDykhta) [reply src/components/src/side-panel/common/types.ts:19]: As I remember it caused ts issues with related components, as something was using fields from KeplerDataset, not available in MiniDataset. I'll check again.

Comment (heshan0131) [reply src/components/src/side-panel/common/types.ts:19]: If that's case, we should add `fields: MiniField[]` to `MiniDataset`

**Pull request merged**